jax/interpreters/pxla.py

```diff
@@ -511,14 +538,26 @@ def __init__(self,
     self.sharding_spec = sharding_spec
     self.indices = indices
     self._npy_value = None
+    self._one_replica_buffer_indices = None
     if not core.skip_checks:
       assert type(aval) is ShapedArray
 
+  @property
+  def one_replica_buffer_indices(self):
+    if self._one_replica_buffer_indices is None:
+      seen_buffer_indices = []
```

Comment:
Rename this `one_replica_indices` or something, it's really a subset of all seen indices